sklearn/qda.py

```diff
@@ -78,159 +72,7 @@ class QDA(BaseEstimator, ClassifierMixin):
-        norm2 = np.array(norm2).T   # shape = [len(X), n_classes]
-        u = np.asarray([np.sum(np.log(s)) for s in self.scalings_])
-        return (-0.5 * (norm2 + u) + np.log(self.priors_))
-
-    def decision_function(self, X):
-        """Apply decision function to an array of samples.
-
-        Parameters
-        ----------
-        X : array-like, shape = [n_samples, n_features]
-            Array of samples (test vectors).
-
-        Returns
-        -------
-        C : array, shape = [n_samples, n_classes] or [n_samples,]
-            Decision function values related to each class, per sample.
-            In the two-class case, the shape is [n_samples,], giving the
-            log likelihood ratio of the positive class.
-        """
-        dec_func = self._decision_function(X)
-        # handle special case of two classes
-        if len(self.classes_) == 2:
-            return dec_func[:, 1] - dec_func[:, 0]
-        return dec_func
-
-    def predict(self, X):
-        """Perform classification on an array of test vectors X.
-
-        The predicted class C for each sample in X is returned.
-
-        Parameters
-        ----------
-        X : array-like, shape = [n_samples, n_features]
-
-        Returns
-        -------
-        C : array, shape = [n_samples]
-        """
-        d = self._decision_function(X)
-        y_pred = self.classes_.take(d.argmax(1))
-        return y_pred
-
-    def predict_proba(self, X):
-        """Return posterior probabilities of classification.
-
-        Parameters
-        ----------
-        X : array-like, shape = [n_samples, n_features]
-            Array of samples/test vectors.
-
-        Returns
-        -------
-        C : array, shape = [n_samples, n_classes]
-            Posterior probabilities of classification per class.
-        """
-        values = self._decision_function(X)
-        # compute the likelihood of the underlying gaussian models
-        # up to a multiplicative constant.
-        likelihood = np.exp(values - values.max(axis=1)[:, np.newaxis])
-        # compute posterior probabilities
-        return likelihood / likelihood.sum(axis=1)[:, np.newaxis]
-
-    def predict_log_proba(self, X):
-        """Return posterior probabilities of classification.
-
-        Parameters
-        ----------
-        X : array-like, shape = [n_samples, n_features]
-            Array of samples/test vectors.
-
-        Returns
-        -------
-        C : array, shape = [n_samples, n_classes]
-            Posterior log-probabilities of classification per class.
-        """
-        # XXX : can do better to avoid precision overflows
-        probas_ = self.predict_proba(X)
-        return np.log(probas_)
+        warnings.warn("qda.QDA has been moved to"
+                      " discriminant_analysis.QuadraticDiscriminantAnalysis"
+                      " in 0.16 and will be removed in 0.18.", DeprecationWarning)
+        super().__init__()
```

Comment:
This is the source of the trouble. For python 2.x you need to pass self and the class of self explicitly. Also, you need to give it the init arguments (just copy and paste).